### ISSUE TITLE
enable branch-based editions

### ISSUE THREAD

**Issue by tayloraswift** [tayloraswift/swift-unidoc#212]
right now, editions are very tag-centric, which presents difficulties when trying to iterate on documentation writing.

**Comment by tayloraswift** [comment]
it looks like we have room in the schema for editions that are neither releases nor prereleases

```
swiftinit-rs [direct: primary] unidoc> db.runCommand({find: 'Editions', filter: { R: { $exists: false } }, limit: 10})
{
  cursor: { firstBatch: [], id: Long('0'), ns: 'unidoc.Editions' },
  ok: 1,
  '$clusterTime': {
    clusterTime: Timestamp({ t: 1715106251, i: 11 }),
    signature: {
      hash: Binary.createFromBase64('AAAAAAAAAAAAAAAAAAAAAAAAAAA=', 0),
      keyId: Long('0')
    }
  },
  operationTime: Timestamp({ t: 1715106251, i: 11 })
}

```

**Comment by tayloraswift** [comment]
implemented by #213
